### DISCUSSION TITLE
Separate sessions of the same project?

### DISCUSSION THREAD

**Discussion by FinYang** [posit-dev/positron#4657] [Feedback] [answered]
Is it possible to have separate sessions/windows of the same project running at the same time? 

This was what `Session` -> `New session` in RStudio does, but in Positron, if I try `New Window`, the new window has no project open. If I try to open the same folder (in a new window), it will just focus the original Positron session, instead of opening a new one.

Sometimes I want to work on the same project but in different folders/sets of files while one session is running. Having a separate session allows me to have a separate console, and also keep the format/layout/opened files unchanged for the original session.

**Comment by jmcphers** [comment, marked as answer]
Yes, in Positron these are called "workspaces". You can't have two identical workspaces open at once, but you _can_ create two workspaces that use the same project folder.

<img width="613" alt="image" src="https://github.com/user-attachments/assets/0bf16932-79ec-4167-8e3b-7c5e0f3429b4">

With your project open, select this command (_Duplicate as Workspace in New Window_). It will create a new workspace against the same project you currently have open, then open a new window in that workspace. Does that do what you want?

**Reply by FinYang** [reply]
Perfect! Thank you so much

**Reply by jrosell** [reply]
Thanks @jmcphers. It's strange that if one try to open a new window and select the project in the new window, positron focus the previous window instead of duplicating the workspace. Do you know if this is the expected behaviour?

**Reply by jmcphers** [reply]
Yes it is (it's inherited behavior from VS Code).

**Reply by juliasilge** [reply]
Indeed that is the expected behavior, yep. We inherit this behavior from upstream and you [might be interested in reading some of the ways to open the same directory twice](https://stackoverflow.com/questions/49707703/open-the-same-directory-twice).

**Reply by jrosell** [reply]
Thanks @juliasilge. The only solution that it seems to work in positron right now is "Open the command panel (ctrl+shift+p), enter dupl, and select Workspaces: Duplicate Workspace in New Window" as @jmcphers already answered here. I wish we could have a new "Duplicate folder in New window..." option in positron, because sometimes I need more than one console and I think it would be faster and easier. 

PD: Just find "File > Duplicate Workspace" LOL

**Reply by juliasilge** [reply]
Would you say the reason you want to do this is that you need multiple consoles? We have that tracked in https://github.com/posit-dev/positron/issues/3173.

**Reply by jrosell** [reply]
Imagine I'm running some tune_grid... But I don't want to wait. Then I want to duplicate the workspace and, within the same project, do some EDA, post modeling analysis or try some new preprocessing, etc

**Reply by juliasilge** [reply]
For folks who are interested in this feature, please check out this announcement from @dhruvisompura about how to get started with an experimental version of multiple console sessions: <https://github.com/posit-dev/positron/discussions/6712>
